frontend/src/components/BucketListModal.tsx

```diff
@@ -51,8 +41,8 @@ const ListModal = ({show, handleClose, setHasChanged}: ListModalProps) => {
                         <Form.Control
                             required
                             type="text"
-                            value={name}
-                            onChange={(e) => setName(e.target.value)}
+                            value={bucketListItem.name}
+                            onChange={(e) => setBucketListItem({...bucketListItem, name: e.target.value})}
                             autoFocus
                         />
```

Comment:
Do you really need this autoFocus attribute? Autofocus can dump screenreader users into your form with zero context, leaving them lost and confused on your page. When using autoFocus you should always keep in mind a11y-rules. https://www.boia.org/blog/accessibility-tips-be-cautious-when-using-autofocus